### ISSUE TITLE
Offer updates/guidance on "PATH" to zq tools included with Brim

### ISSUE THREAD

**Issue by philrz** [brimsec/brim#1043]
We already bundle the `zq` binary with Brim, and may include more in the future (https://github.com/brimsec/brim/issues/1038). For Brim desktop users who want to work at the command line, it may be ideal for them to that particular set of zq tools, both as a convenience and to avoid potential hazards if they download a separate zq binary bundle that happens to be incompatible with their installed Brim version at the data/API layer(s). While we could construct full pathnames to the binaries where possible (https://github.com/brimsec/brim/issues/580) users would likely find it even more convenient if the tools were in the `PATH` variable appropriate for their OS platform.

If we wanted to address this, it seems like separate approaches might be necessary for each platform.

1. On Windows, I've known installers in the past that prompted a user to ask if they wanted their `%PATH%` automatically updated to include a directory. I don't know the technical feasibility of this with our Electron installer, but given that updating the `%PATH%` manually on Windows is kind of a pain IMO, this might be a great convenience.

2. On Linux/Mac, I'm assuming that the diverse options of different shells (`bash`, `zsh`, `tcsh`, etc.) and different places `$PATH` could be maintained (`.profile`, `.bashrc`, etc.) means that it's probably not feasible to automatically update this for the user. Perhaps the best we could do is prominently display the full pathname to `zdeps/` at appropriate locations in the app along with the suggestion that they add it to their `$PATH`. One obvious place to do this would be in the "Copy for curl" modal (and if we did this, perhaps we could skip https://github.com/brimsec/brim/issues/580) but it might also be helpful to pop up the information in a modal the very first time they run Brim (one of those notices where they can check "Do not display again" etc.)

**Comment by cshanahan** [comment]
Hello,
Is there a detailed installation document that explains which packages are required for Brim to function correctly, and how to handle PATH, Preferences, and permissions? Brim isn't working for me but before I open an issue I'd like to make sure I installed the app correctly, and I suspect my problem has something to do with either PATH for zq or permissions.

![image](https://user-images.githubusercontent.com/4239381/93792971-b6dabf00-fc25-11ea-83d9-cf2b6a9097f1.png)

The [Downloads page](https://www.brimsecurity.com/download/) includes links for both Brim and zq, but no other tips on how to properly install and configure the application. The Wiki in this repo includes a [Troubleshooting section](https://github.com/brimsec/brim/wiki/Troubleshooting), which I read, but no installation document.
In my case I am using an RPM-based Linux distro, so I did the following:

1. Installed Brim v0.17.0 via rpm
2. Downloaded zq v0.21.0 Zip file for Linux
3. Unzipped zq to a directory and added the zq directory to PATH
4. Launch Brim > The app seems to start without error however the app will not parse any PCAP files

Thank you

**Comment by philrz** [comment]
@cshanahan: Sorry to hear you're having trouble. For starters, I'll suggest that if you're up for it, it might make sense to join our public Slack via https://www.brimsecurity.com/join-slack/ , as I expect back & forth debug might be easier via that medium. If for some reason that's not feasible for you we can certainly keep debugging via GitHub Issues or email at support@brimsecurity.com.

To get things started, though, I'll respond to your initial inquiries here. There's not a detailed installation document because users have traditionally not needed one. The fact it didn't work seamlessly for you out-of-the-box is definitely something we'll want to dig into, then. While the items in that Preferences menu are there for advanced users seeking to customize certain functionality, it's by no means necessary for the initial use case of importing pcaps and seeing Zeek logs for them. Also, while zq is available as a separate/standalone set of CLI tools, you don't need to install it or put it in the `$PATH` like you described. There's a `zdeps/` within the hierarchy of what gets installed with Brim, and that includes the `zqd` binary that acts as the local backend for the query language/engine, and also the Zeek executable & "runner" script for invoking it with pcaps. The app is hard-wired by default to find those tools in that directory and invoke them as necessary.

In conclusion, since you say the app launched but "will not parse any PCAP files", the next step will be to understand more about that. Does it pop up any kind of error message? Does it just show a blank screen with "No result data"? If so, is there perhaps a red exclamation point in the lower-right corner of the window after the attempted pcap import? That might reveal  more detail on what might have gone wrong. And if not, I have other ideas of places we can look for debug. So, once again, if you can join Slack via https://www.brimsecurity.com/join-slack/ , feel free to speak up in the `#brim` channel or message me directly `@phil`, or we can continue to work through it here or in a new issue if you'd like to open one.
